lib/puppet/provider/zypprepo/inifile.rb

```diff
@@ -71,11 +71,12 @@ def self.prefetch(resources)
   # @param conf [String] Configuration file to look for directories in.
   # @param dirs [Array<String>] Default locations for zypper repos.
   # @return [Array<String>] All present directories that may contain zypper repo configs.
-  def self.reposdir(conf = '/etc/zypp/zypper.conf', dirs = ['/etc/zypp/repos.d'])
+  def self.reposdir(conf = '/etc/zypp/zypp.conf', dirs = ['/etc/zypp/repos.d'])
```

Comment:
how does this change work (I've zero knowledge about suse)? Did we manage /etc/zypp/zypper.conf in the past and now /etc/zypp/zypp.conf? Was this a typo and it should have been /etc/zypp/zypper.conf all the time / is there now a /etc/zypp/zypp.conf that we should clean up?

I just want to understand if this could create any issues for people that upgrade the module.

Reply:
It should have been /etc/zypp/zypp.conf all the time. It was an error in my first type/provider implementation.
We only read from /etc/zypp/zypp.conf to get a reposdir config item in the main section. We don't write to /etc/zypp/zypp.conf at all.